aten/src/ATen/native/cpu/batch_norm_kernel.cpp

```diff
@@ -192,27 +189,44 @@ batch_norm_cpu_collect_stats_contiguous_impl(
   // parallel dim reduce on 'channel'
   at::parallel_for(0, n_channel, 1, [&](int64_t begin, int64_t end) {
     for (const auto c : c10::irange(begin, end)) {
-      // compute mean per input
-      accscalar_t sum = 0;
-      for (const auto n : c10::irange(n_batch)) {
-        for (const auto i : c10::irange(image_size)) {
-          auto offset = n * n_channel * image_size + c * image_size + i;
-          sum += input_data[offset];
+    // compute mean per input
+      scalar_t sum_val = 0;
+      Vec sum_val_vec = Vec(scalar_t(0));
+      for (int64_t n = 0; n < n_batch; n++) {
+        const scalar_t* input_ptr = input_data + n * n_channel * image_size + c * image_size;
+        int64_t d = 0;
+        for (; d < image_size - (image_size % Vec::size()); d += Vec::size()) {
+          Vec data_vec = Vec::loadu(input_ptr + d);
+          sum_val_vec += data_vec;
+        }
+        for (; d < image_size; d++) {
+          sum_val += input_ptr[d];
         }
       }
-      scalar_t mean = sum / N;
+      sum_val += vec_reduce_all([](Vec& x, Vec& y) { return x + y; }, sum_val_vec);
```

Comment:
`vec_reduce_all(const Op&, const Vectorized<scalar_t>&)` here is already the `fast` version: https://github.com/pytorch/pytorch/tree/main/aten/src/ATen/cpu#L82C17-L82C31, it goes into a vectorized horizontal reduce of SIMD

`vec_reduce_all(const Op&, const Vectorized<scalar_t>&), int64_t` is a slow version when the size < vector length.